generator/src/main/java/org/corfudb/generator/LongevityApp.java

```diff
@@ -0,0 +1,168 @@
+package org.corfudb.generator;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.generator.operations.CheckpointOperation;
+import org.corfudb.generator.operations.Operation;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by rmichoud on 7/25/17.
+ */
+
+/**
+ * This Longevity app will stress test corfu using the Load
+ * generator during a given duration.
+ *
+ * I use a blocking queue in order to limit the number of operation
+ * created at any time. Operations as created and consumed as we go.
+ *
+ */
+@Slf4j
+public class LongevityApp {
+    private long durationMs;
+    private String configurationString;
+    private boolean checkPoint;
+    BlockingQueue<Future> appsFutures;
+    BlockingQueue<Operation> operationQueue;
+    CorfuRuntime rt;
+    State state;
+
+    ExecutorService taskProducer;
+    ScheduledExecutorService checkpointer;
+    ExecutorService workers;
+
+    // How much time we live the application hangs once the duration is finished
+    // and the application is hanged
+    final int applicationTimeoutInMs = 20000;
+
+    final int queueCapacity = 1000;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this final field static too. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1170)